### PR TITLE
can not list tables after test connect success and explore button is always disable in query result view

### PR DESCRIPTION
**Pull request by cncowboy**
### CATEGORY

Choose one

- [x ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

1、display tables after datasource test connecting ok
2、enable explore button after query result ok, then create a chart after click explore button

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -1727,7 +1727,8 @@ def testconn(self):
 
             with closing(engine.connect()) as conn:
                 conn.scalar(select([1]))
-                return json_success('"OK"')
+                return json_success(json.dumps(engine.table_names(), indent=4))
```
Comment (etr2460) [inline superset/views/core.py]: It's intentional that we don't dump the table names when testing the connection. This api shouldn't be used for anything more than making sure the connection to a db works

File: superset/assets/src/SqlLab/reducers/sqlLab.js [hunk 2]
```diff
@@ -318,6 +318,7 @@ export default function sqlLabReducer(state = {}, action) {
       const databases = {};
       action.databases.forEach((db) => {
         databases[db.id] = db;
+        databases[db.id]['allows_subquery'] = true;
```
Comment (etr2460) [inline superset/assets/src/SqlLab/reducers/sqlLab.js]: does every db allow subqueries? Shouldn't this be coming from the backend?

**Review by etr2460 — COMMENTED** [review]
could you attach screenshots showing the behavior before and after your change? Thanks!

File: superset/views/core.py (hunk 1 above)
Comment (mistercrunch) [reply superset/views/core.py]: Yes that's what it did historically for no really good reason (returning table names). Returning a 200 "OK"seems right.

File: superset/assets/src/SqlLab/reducers/sqlLab.js (hunk 2 above)
Comment (cncowboy) [reply superset/assets/src/SqlLab/reducers/sqlLab.js]: maby not all,

**Pull request closed without merging**